graphbolt/src/random_engine.cc

```diff
@@ -0,0 +1,125 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file random_engine.cc
+ * @brief Source file of random engine.
+ */
+
+#include "random_engine.h"
+
+#include <algorithm>
+#include <unordered_set>
+#include <vector>
+
+namespace graphbolt {
+namespace sampling {
+
+template <typename T>
+T RandomEngine::RandInt(T lower, T upper) {
```

Comment:
I think we do need to think about the layout of files in this directory as more and more files are being added for different usages.